### PR TITLE
[Fluent 2 iOS] Card Colors Update

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

The Card colors were updated to match fluent 2 specs.

### Verification

The cards were tested on the demo app.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_light | after_light |
| before_dark | after_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1065)

### CODE REVIEW

File: ios/FluentUI/Card/CardView.swift [hunk 1]
```diff
@@ -409,19 +418,19 @@ open class CardView: UIView {
     private func setupColors() {
         switch colorStyle {
         case .appColor:
```
Comment (huwilkes) [inline ios/FluentUI/Card/CardView.swift:420]: Looks like we lost the variation that had the brand colors. Can you follow up with design, and if that was intentional, create a work item to track the removal of the unused styles? Or at least to condense the shared logic. I guess there's still a variation that uses brand color for the icon.

**Review by huwilkes — APPROVED** [review]

File: ios/FluentUI/Card/CardView.swift (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI/Card/CardView.swift:420]: The brand colors were updated on the last commit. Design omitted them on the Figma but they gave me the right tokens this morning.

File: ios/FluentUI/Card/CardView.swift [hunk 2]
```diff
@@ -409,19 +418,19 @@ open class CardView: UIView {
...
-            primaryLabel.textColor = UIColor(light: .black, dark: Colors.gray100)
-            secondaryLabel.textColor = UIColor(light: Colors.gray600, dark: Colors.gray400)
-            iconView.tintColor = UIColor(light: Colors.gray600, dark: Colors.gray500)
+            primaryLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground1])
+            secondaryLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground2])
+            iconView.tintColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.brandForeground1])
             if let window = window {
```
Comment (huwilkes) [inline ios/FluentUI/Card/CardView.swift:424]: Also, since you're pulling the tokens from the theme, you don't need the window anymore

**Comment by edjamesmsft** [comment]
Nit: could you update the title with  [Fluent 2 iOS] at the beginning? No huge deal though!

**Comment by edjamesmsft** [comment]
Is this actually how we're showing a card is selected, just updating the color of the icon but not the card itself? That seems like a big step back for accessibility. Could you double check with design?

**Comment by laminesm** [comment]
> Is this actually how we're showing a card is selected, just updating the color of the icon but not the card itself? That seems like a big step back for accessibility. Could you double check with design?

I think those are the cards with brand colors, not selected cards.  They weren't initially part of the design but Ian gave me these tokens today with brand colors only for the icon.

**Pull request merged**